git/clone/bin/git-clone

```diff
@@ -201,3 +201,52 @@ if [[ "${PRESERVE_GIT_DIR}" == "false" ]]; then
     git config --file .gitmodules --get-regexp path | awk '{ print $2 }' | xargs -I {} rm -rf {}/.git
   fi
 fi
+
+if [ ! -d "$PATCHES_DIR" ]; then
+  echo ""
+  echo "No patches to apply"
+  exit 0
+fi
+
+for file in "$PATCHES_DIR"/*; do
+  echo ""
+  filename=$(basename "$file")
+  if [ -d "$file" ]; then
+    echo "Patch file $filename is a directory. Skipping..."
+  elif [ -f "$file" ]; then
+    if [ "$filename" == "$commit_sha" ]; then
+      echo "Patch file $filename matches checked out sha."
+      echo "Generating dynamic git-patch task..."
+
+      echo $file > $RWX_VALUES/patch-file
+
+      cat << EOF > $MINT_DYNAMIC_TASKS/patch.yml
+- key: git-patch
+  use: git-clone 
+  run: |
+    # Apply the patch
+    echo "Applying patch file \${PATCH_FILE}..."
+    cd "\${CHECKOUT_PATH}"
+
+    git apply --stat --summary "\${PATCH_FILE}"
+    git apply "\${PATCH_FILE}"
+
+    # Re-export env vars
+    PATCH_GIT_COMMIT_SHA="\${RWX_GIT_COMMIT_SHA}+patch"
+    printf "%s" "\${PATCH_GIT_COMMIT_SHA}" >> "\${MINT_CACHE_KEY_EXCLUDED_ENV}/MINT_GIT_COMMIT_SHA"
+    printf "%s" "\${PATCH_GIT_COMMIT_SHA}" >> "\${MINT_CACHE_KEY_EXCLUDED_ENV}/RWX_GIT_COMMIT_SHA"
+
+    # Upload an artifact
+    cp \${PATCH_FILE} \$RWX_ARTIFACTS/\$(basename "\$PATCH_FILE").patch
```

Comment:
This concerns me a little because we don't really know what they might have staged locally and whether anything is sensitive (i.e. tokens, etc), but maybe it's already discussed elsewhere 🤷‍♂️